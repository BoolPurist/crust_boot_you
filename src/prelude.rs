use std::io;

pub use crate::constants;
pub use anyhow::Context;

pub type IoResult<T = ()> = Result<T, io::Error>;
pub type AppError = anyhow::Error;
pub type AppResult<T = ()> = Result<T, AppError>;

pub type ReturnToUser = AppResult<String>;
pub type PathResult = AppResult<PathBuf>;
pub type AppIoResult<T = ()> = Result<T, AppIoError>;

pub use crate::app_traits::file_manipulator::FileManipulator;
pub use crate::app_traits::path_provider::PathProvider;
pub use crate::file_management::AppIoError;
pub use crate::NotEmptyText;
pub use std::path::{Path, PathBuf};

#[cfg(test)]
pub use mockall::automock;
