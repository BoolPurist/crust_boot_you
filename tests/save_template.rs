mod common;
use common::insta_utils;
use common::prelude::*;
use crust_boot_you::{
    app_traits::{file_manipulator::DevOsFileManipulator, path_resolver::DevPathResolver},
    cli::{AbsoluteExistingPath, SaveTemplateCli, TemplateCliArg},
    handle_commands, DevPathProvider,
};

#[test]
#[named]
fn save_file_new_template() {
    let setup = TestSetup::new(actual_expected!());
    let tmp_path = setup.path_to_temp();
    let path_resol = DevPathResolver::new(tmp_path.to_path_buf());
    let path_provider = DevPathProvider::new(tmp_path.to_path_buf());
    let file_sys = DevOsFileManipulator::new(tmp_path);

    let given_name = ValidTemplateName::new("a".to_string()).unwrap();
    let given_path = AbsoluteExistingPath::new(PathBuf::from("new.txt"), &path_resol).unwrap();
    let arguments: SaveTemplateCli = TemplateCliArg::new(given_name, given_path).into();
    let output =
        handle_commands::handle_save_template(path_provider, file_sys, &arguments).unwrap();
    insta::with_settings!({ filters => insta_utils::filter_random_tmp_folder_name() }, { insta::assert_display_snapshot!(output) });

    setup.assert_with_expected();
}
