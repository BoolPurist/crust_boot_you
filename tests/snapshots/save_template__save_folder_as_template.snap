---
source: tests/save_template.rs
expression: output
---
Created template with name (some hello project) from the folder "/[TMP_ROOT]/lang/go"
